### PR TITLE
Added support for "enums"

### PR DESCRIPTION
**Pull request by glooms**
Enums in the API are represented as strings with a list of possible values.
These will be interpreted as `type SomeType string` with two functions: `String()` and `MarshalText()`

When creating structs these types will be required, but, from a user a perspective they can just use strings as before.

### CODE REVIEW

**Comment by glooms** [comment]
If used implicitly, like so:
```
HyperCubeDef{
  ...
  Mode: "DATA_MODE_PIVOT",
  ...
}
```
the changes will not affect the user other than that they will receive feedback regarding erroneous values. However, if the user instead writes something like this:
```
mode := "DATA_MODE_PIVOT"
HyperCubeDef{
  ...
  Mode: mode
  ...
}
```
it will now fail with an error message like: cannot use type `string` as type `enigma.SomeType`.

The fix is easy for the user (let go do an implicit conversion or cast to the specific type) but it might not be obvious. I'm wondering how and where to document this.

File: examples/data/hypercubes/pivot/hypercube-pivot.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func main() {
 					Def: "Sum(Value)",
 				},
 			}},
-			Mode:                "EQ_DATA_MODE_PIVOT",
+			Mode:                "DATA_MODE_PIVOT",
```
Comment (peol) [inline examples/data/hypercubes/pivot/hypercube-pivot.go]: Why is this changed?

File: schema/README.md [hunk 2]
```diff
@@ -14,3 +14,15 @@ If a version is not specified the script will default to the latest published ve
 ```bash
 ACCEPT_EULA=<yes/no> ./schema/generate.sh
 ```
+
+## Adding Allowed Arguments for "Enumtypes"
+
+This section if for you who are developing and want to extend the set of allowed arguments for a property a given property.
```
Comment (peol) [inline schema/README.md]: This sentence sounds weird :)

File: schema/README.md [hunk 3]
```diff
@@ -14,3 +14,15 @@ If a version is not specified the script will default to the latest published ve
...
+Let's take an example working with `HyperCubeDef`. This definition has a property called `qMode` which is represented by an enum called `NxHypercubeMode`.
```
Comment (peol) [inline schema/README.md]: Link to https://core.qlik.com/services/qix-engine/apis/qix/definitions/#hypercubedef ?

We should also consider adding an enum page to the core.qlik.com docs.

File: schema/README.md [hunk 4]
```diff
@@ -14,3 +14,15 @@ If a version is not specified the script will default to the latest published ve
...
+
+In `enigma.go`, the allowed arguments will be restricted based on the different values that `NxHypercubeMode` can have. So if you want to extend the allowed arguments you can use:
```
Comment (peol) [inline schema/README.md]: Isn't it `enigma-go`? Anyway, I don't think you necessarily have to mention it in its own documentation that explicitly :)

File: schema/README.md (hunk 3 above)
Comment (glooms) [reply schema/README.md]: Sounds good to me!

File: examples/data/hypercubes/pivot/hypercube-pivot.go (hunk 1 above)
Comment (glooms) [reply examples/data/hypercubes/pivot/hypercube-pivot.go]: According to the API specification "EQ_DATA_MODE_PIVOT" is not allowed as a Mode for a HyperDef, it should be "DATA_MODE_PIVOT".

**Comment by glooms** [comment]
Closing while we wait for @wennmo and @gabbaxx

**Comment by wennmo** [comment]
@gabbaxx Can you take a look at this PR?

**Comment by glooms** [comment]
I agree with @gabbax. I think that dynamic checks aren't very useful and that we shouldn't be too restrictive regarding what arguments we allow. I say we close this PR and include the information we can now gather about error codes in #78 instead.
